### PR TITLE
Cache namespaces per cluster and per token

### LINKED ISSUE #6037: Multicluster - Token per cluster
Associate a token per user per cluster. Kiali can assume that the user credentials will remain the same across clusters but the backend will still need to authenticate with those credentials to each kube cluster to offer a SSO experience with multicluster.

### PR DESCRIPTION
**Pull request by nrfox**
### Describe the change

Caches namespaces per token and per cluster rather than simply per token.

### Steps to test the PR

Regression testing.

### Automation testing

Unit tests added. Unable to fully cover with e2e tests since token per cluster is not a supported deployment yet.

### Issue reference

Fixes #6037

### CODE REVIEW

File: business/istio_validations.go [hunk 1]
```diff
@@ -553,18 +553,14 @@ func (in *IstioValidationsService) fetchNonLocalmTLSConfigs(mtlsDetails *kuberne
...
 
-	kubeCache := kialiCache.GetKubeCaches()[cluster]
-	if kubeCache == nil {
-		return
-	}
-
-	userClient := in.userClients[cluster]
-	if userClient == nil {
+	cfg := config.Get()
+	// TODO: Handle multi-primary instead of only using home cluster.
```
Comment (jshaughn) [inline business/istio_validations.go:557]: Is there an issue for this?  We do claim MP support at this point.

File: kubernetes/cache/cache.go [hunk 2]
```diff
@@ -202,5 +204,78 @@ func (in *kialiCacheImpl) IsAmbientEnabled(cluster string) bool {
...
+	c.namespacesLock.Lock()
+	defer c.namespacesLock.Unlock()
+
+	namespacesByCluster := make(map[string][]models.Namespace)
+	for _, namespace := range namespaces {
+		namespacesByCluster[namespace.Cluster] = append(namespacesByCluster[namespace.Cluster], namespace)
+	}
+
+	for cluster, clusterNamespaces := range namespacesByCluster {
+		key := namespacesKey{cluster: cluster, token: token}
```
Comment (jshaughn) [inline kubernetes/cache/cache.go:257]: Any need to validate the token here, check for empty, before it gets used in a map key?

**Review by jshaughn — APPROVED** [review]
Visual inspection of logic looks good.  Good, distinct error messages.  A couple of minor, non-blocking, comments

File: business/istio_validations.go (hunk 1 above)
Comment (nrfox) [reply business/istio_validations.go:557]: This is one of those things that should get cleaned up with moving things to the mesh page. I'll create a separate issue to capture that list.

Comment (nrfox) [reply business/istio_validations.go:557]: Created: https://github.com/kiali/kiali/issues/7203

**Pull request merged**